### PR TITLE
Add support for vault access token. Closes #81

### LINKED ISSUE #81: Enable Test-TppToken to access vault in secret store
Enable Test-TppToken to access vault in secret store

Currently Test-TppToken allows you to test an individual token is still valid using the 'Authorize/Verify' call to TPP. Either a PSCredential type access token or a PSCustomObject containing the access token must be supplied.

This could be expanded to be able to get the access token from a vault using the PowerShell Secret Management module(s) using a 'VaultAccessTokenName' parameter.

### PR DESCRIPTION
**Pull request by wilddev65**
Adds support for parameter 'VaultAccessTokenName' which can be used to validate access tokens stored in VenafiPS vault.

Fixes #81

### CODE REVIEW

File: VenafiPS/Public/Test-TppToken.ps1 [hunk 1]
```diff
@@ -76,15 +85,19 @@ function Test-TppToken {
                 }
             }
         )]
-        [Alias('Server')]
-        [string] $AuthServer,
+        [Alias('ServerUrl')]
+        [string] $Server,
```
[the remark this answers is not in the document]
Comment (wilddev65) [reply VenafiPS/Public/Test-TppToken.ps1:89]: Will change this back to AuthServer

File: VenafiPS/Public/Test-TppToken.ps1 [hunk 2]
```diff
@@ -120,16 +133,49 @@ function Test-TppToken {
...
+                $tokenSecret = Get-Secret -Name $VaultAccessTokenName -Vault 'VenafiPS' -ErrorAction SilentlyContinue
+                if ( -not $tokenSecret ) {
+                    throw "'$VaultAccessTokenName' secret not found in vault VenafiPS."
+                }
+
+                # check if metadata was stored
+                $secretInfo = Get-SecretInfo -Name $VaultAccessTokenName -Vault 'VenafiPS' -ErrorAction SilentlyContinue
+
+                if ( $secretInfo.Metadata.Count -gt 0 ) {
+                    $params.Server = $secretInfo.Metadata.Server
```
[the remark this answers is not in the document]
Comment (wilddev65) [reply VenafiPS/Public/Test-TppToken.ps1:166]: Fixed

File: VenafiPS/Public/Test-TppToken.ps1 [hunk 3]
```diff
@@ -5,13 +5,17 @@ Test if a Tpp token is valid
...
+.PARAMETER Server
 Auth server or url, venafi.company.com or https://venafi.company.com.
+This will be used to access vedauth for token-based authentication.
 If just the server name is provided, https:// will be appended.
 
 .PARAMETER AccessToken
 Access token retrieved outside this module.  Provide a credential object with the access token as the password.
 
+.PARAMETER VaultAccessTokenName
+Name of the SecretManagement vault entry for the access token; the name of the vault must be VenafiPS.
```
[the remark this answers is not in the document]
Comment (wilddev65) [reply VenafiPS/Public/Test-TppToken.ps1:17]: Added blurb for this

File: VenafiPS/Public/Test-TppToken.ps1 [hunk 4]
```diff
@@ -76,15 +85,19 @@ function Test-TppToken {
...
+        [string] $Server,
 
         [Parameter(Mandatory, ParameterSetName = 'AccessToken', ValueFromPipeline)]
         [PSCredential] $AccessToken,
 
         [Parameter(Mandatory, ParameterSetName = 'TppToken', ValueFromPipeline)]
         [pscustomobject] $TppToken,
 
+        [Parameter(Mandatory, ParameterSetName = 'VaultAccessToken')]
+        [Parameter(ParameterSetName = 'AccessToken')]
```
[the remark this answers is not in the document]
Comment (wilddev65) [reply VenafiPS/Public/Test-TppToken.ps1:98]: Removed

**Review by gdbarron — APPROVED** [review]

**Review by denvenafi — APPROVED** [review]

**Pull request merged**